machine.go

```diff
@@ -211,6 +213,13 @@ func (h *Headscale) GetMachineByID(id uint64) (*Machine, error) {
 	return &m, nil
 }
 
+func (h *Headscale) UpdateMachine(m *Machine) error {
```

Comment:
What is this function used for? It seems quite unused / no-op

Reply:
It is quite used:
https://github.com/kradalby/headscale/blob/integration-tests-ci/poll.go#L240
https://github.com/kradalby/headscale/blob/integration-tests-ci/poll.go#L286
https://github.com/kradalby/headscale/blob/integration-tests-ci/poll.go#L350
https://github.com/kradalby/headscale/blob/integration-tests-ci/poll.go#L350

It is the discovery (and initial mitigation) of #93